py/mpconfig.h

```diff
@@ -1814,6 +1814,13 @@ typedef double mp_float_t;
 #define MICROPY_OBJ_BASE_ALIGNMENT
 #endif
 
+// A port may explicitly disable the NLR if it is unable to provide a suitable
+// implementation. If so, it must provide nlr_jump, and handle that in some
+// way (i.e. terminate).
+#ifndef MICROPY_NLR
```

Comment:
Do you think disabling this will be useful for more than just a mpy-cross wasm workaround?  I preferred your previous solution, just a simple way to overridde NLR in "emergency" situations.

Reply:
I don't think so (although maybe it's useful when first starting a port?).

I do like that this way is very explicit as to what it's doing -- everything is a no-op except nlr_jump.

The previous solution was a bit awkward in that it relied on the particular interpretation of disabling setjmp.

Reply:
I've re-done this by adding a `MICROPY_NLR_TERMINATE` option. This is how a port (e.g. mpy-cross.wasm) signals that it explicitly wants this disable, but it's overall a much smaller change to nlr.{ch}. The idea is that it mirrors the existing `MICROPY_NLR_SETJMP` which is how a port signals that it explicitly wants the setjmp implementation.